### PR TITLE
Cds 45998 terragrunt ng

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+Here is an example of the YAML for a Terragrunt Rollback step:
+
+```yaml
+            rollbackSteps:
+              - step:
+                  type: TerragruntRollback
+                  name: Terragrunt Rollback_1
+                  identifier: TerragruntRollback_1
+                  spec:
+                    provisionerIdentifier: abc123abc
```
Comment (vlica-harness) [inline docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:770]: for more clarity, this provisionerIdentifier  **abc123abc** should correspond to other terragrunt steps(we do inherit there), should be changed to this: **planinherit123aa1**

**Review by vlica-harness — DISMISSED** [review]

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+```
+
+To add a Terragrunt Destroy step, do the following:
+
+1. In your CD stage **Execution**, click **Add Step**, and then click **Terragrunt Destroy**.
+2. Enter the following Terragrunt Destroy settings.
+
+#### Name
+
+1. In Name, enter a name for the step. Use a name that describes the infrastructure the step plans to provision.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:311]:
```suggestion
* In Name, enter a name for the step. Use a name that describes the infrastructure the step plans to provision.
```

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+
+#### Timeout
+
+1. Enter how long you want Harness to try to complete the step before failing and initiating the stage or step [failure strategy](https://developer.harness.io/docs/platform/pipelines/define-a-failure-strategy-on-stages-and-steps/).
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:315]:
```suggestion
* Enter how long you want Harness to try to complete the step before failing and initiating the stage or step [failure strategy](https://developer.harness.io/docs/platform/pipelines/define-a-failure-strategy-on-stages-and-steps/).
```

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md [hunk 4]
```diff
@@ -0,0 +1,781 @@
...
+You can add a Terragrunt Destroy step to remove any provisioned infrastructure, just like running the `terragrunt run-all destroy` command. See [destroy](https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/#the-run-all-command) from Terragrunt.
+
+1. In **Configuration Type**, select how you want to destroy resources:
+
+- **Inherit From Plan**. Destroy the resources from a Terraform Plan step.
+  - Using the Terragrunt Destroy step with a previous Terragrunt Plan step is the same as using the `terragrunt plan` command with the `-destroy` flag.
+  - To use this Terraform Destroy step with a Terraform Plan step, you must select **Destroy** in the **Command** setting of the **Terraform Plan** step.
+- **Inherit From Apply**. Destroy the resources from a Terraform Apply step.
+  - Using the Terragrunt Destroy step with a previous Terragrunt Apply step is the same as using the `terragrunt apply` command with the `-destroy` flag.
+- **Inline**. Destroy any resources using a Terragrunt script.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:343]:
```suggestion
* In **Configuration Type**, select how you want to destroy resources:

    - **Inherit From Plan**. Destroy the resources from a Terraform Plan step.
        - Using the Terragrunt Destroy step with a previous Terragrunt Plan step is the same as using the `terragrunt plan` command with the `-destroy` flag.
        - To use this Terraform Destroy step with a Terraform Plan step, you must select **Destroy** in the **Command** setting of the **Terraform Plan** step.
     - **Inherit From Apply**. Destroy the resources from a Terraform Apply step.
        - Using the Terragrunt Destroy step with a previous Terragrunt Apply step is the same as using the `terragrunt apply` command with the `-destroy` flag.
    - **Inline**. Destroy any resources using a Terragrunt script.
```

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
+  count="${lookup(local.counts, terraform.workspace, 2)}"  
+  tags {  
+         Name = "${terraform.workspace}"  
+    }  
+}
+```
+
+In the workspace interpolation sequence you can see the count is assigned by applying it to the workspace variable (`terraform.workspace`) and that the tag is applied using the variable also.
+
+1. In **Workspace**, enter the name of the workspace to use.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:539]:
```suggestion
* In **Workspace**, enter the name of the workspace to use.
```

**Review by douglas-j-bothwell — DISMISSED** [review]
Hi Michael, most of my edits are based on the rule "Use a bullet for procedures with only one step." in the Doc Style Guide:
https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide#Procedures

File: docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md (hunk 5 above)
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-advanced/terragrunt/terragrunt-howtos.md:539]: you already made a suggestion on this line, and I can't commit 2 on the same line, so I'll delete this one.

**Review by douglas-j-bothwell — APPROVED** [review]

**Pull request merged**